component/src/main/java/com/celements/blog/article/Article.java

```diff
@@ -445,4 +485,103 @@ public String toString() {
     return "Article [docRef=" + getDocumentReference() + "]";
   }
 
+  String getExtractPlainTextEncoded(String lang, boolean isViewtypeFull, int maxNumChars) {
+    String articleExtract = getExtract(lang, isViewtypeFull, maxNumChars);
+    try {
+      String plainExtract = new PlainTextCommand().convertHtmlToPlainText(articleExtract);
+      return plainExtract.replaceAll("\"", "&quot;");
+    } catch (ConvertToPlainTextException ctpte) {
+      LOGGER.error("Failed to convert article extract to plain text for article {}", articleDocRef,
+          ctpte);
+    }
+    return articleExtract;
+  }
+
+  List<ImageUrlDim> getArticleImagesBySizeAsc(String lang) {
+    ImageUrlExtractor urlManip = new ImageUrlExtractor();
+    List<ImageUrlDim> articleImages = urlManip.extractImagesList(getFullArticle(lang));
+    if (articleImages.isEmpty()) {
+      articleImages = urlManip.extractImagesList(getExtract(lang, false, getMaxNumChars()));
+    }
+    return articleImages;
+  }
+
+  public String getExternalUrl() {
+    String fullName = getModelUtils().serializeRefLocal(articleDocRef);
+    try {
+      return context.getWiki().getExternalURL(fullName, "view", "ref=socialmedia", context);
+    } catch (XWikiException xwe) {
+      LOGGER.error("Exception while trying to get externalURL for doc {}", articleDocRef, xwe);
+    }
+    return context.getWiki().getURL(articleDocRef, "view", context);
+  }
+
+  /**
+   * Returns social media meta tags, respecting the configuration. If the tags are activate the
+   * calculated OpenGraph tags are added. Furthermore if a Twitter account is configured
+   * (e.g. @example), the Twitter tags are added. The Twitter card type is configurable, defaulting
+   * to "summary"
+   *
+   * @param language
+   */
+  public List<MetaTag> getArticleSocialMediaTags(@NotNull String language) {
+    List<MetaTag> metaTags = new ArrayList<>();
+    if (1 == getConfigurationSource().getProperty(BLOG_ARTICLE_SOCIAL_MEDIA_CONF_NAME, 0)) {
```

Comment:
use boolean type instead of integer for clarity if this is the intent. ConfigurationSource does convert correctly from DB integer to java boolean type and reads out true/false from properties file.